File: server/datasets/java_source-code/Multiple-G3/renamed_36_dsu.java
public class DSU_0 {
    public static void main(String[] args) {
        DSU_0 dsu = new DSU_0(10);
        dsu.u_0(1, 2);
    }
    int[] p_0;
    int[] r_0;
    public DSU_0(int tot_0) {
        p_0 = new int[tot_0];
        r_0 = new int[tot_0];
        for (int idx = 0; idx < tot_0; idx++) {
            p_0[idx] = idx;
            r_0[idx] = 0;
        }
    }
    public int f_0(int x_0) {
        if (p_0[x_0] != x_0) {
            p_0[x_0] = f_0(p_0[x_0]);
        }
        return p_0[x_0];
    }
    public void u_0(int a_0, int b_0) {
        int ra_0 = f_0(a_0);
        int rb_0 = f_0(b_0);
        if (ra_0 != rb_0) {
            if (r_0[ra_0] < r_0[rb_0]) {
                p_0[ra_0] = rb_0;
            } else if (r_0[ra_0] > r_0[rb_0]) {
                p_0[rb_0] = ra_0;
            } else {
                p_0[rb_0] = ra_0;
                r_0[ra_0]++;
            }
        }
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_g3/renamed_36_dsu.py
class DSU_0:
    def __init__(self, n_0):
        self.p_0 = list(range(n_0))
        self.rk_0 = [0] * n_0

    def find_0(self, x_0):
        if self.p_0[x_0] != x_0:
            self.p_0[x_0] = self.find_0(self.p_0[x_0])
        return self.p_0[x_0]

    def union_0(self, x_0_a, x_0_b):
        ra_0 = self.find_0(x_0_a)
        rb_0 = self.find_0(x_0_b)
        if ra_0 != rb_0:
            if self.rk_0[ra_0] < self.rk_0[rb_0]:
                ra_0, rb_0 = rb_0, ra_0
            self.p_0[rb_0] = ra_0
            if self.rk_0[ra_0] == self.rk_0[rb_0]:
                self.rk_0[ra_0] += 1
            return True
        return False
